docs/topics/auth/customizing.txt

```diff
@@ -451,33 +451,10 @@ different User model.
     currently active User model -- the custom User model if one is specified, or
     :class:`~django.contrib.auth.models.User` otherwise.
 
-    When you define a foreign key or many-to-many relations to the User model,
-    you should specify the custom model using the :setting:`AUTH_USER_MODEL`
-    setting. For example::
+    .. versionchanged:: 1.11
 
-        from django.conf import settings
-        from django.db import models
-
-        class Article(models.Model):
-            author = models.ForeignKey(
-                settings.AUTH_USER_MODEL,
-                on_delete=models.CASCADE,
-            )
-
-    When connecting to signals sent by the ``User`` model, you should specify
-    the custom model using the :setting:`AUTH_USER_MODEL` setting. For example::
-
-        from django.conf import settings
-        from django.db.models.signals import post_save
-
-        def post_save_receiver(sender, instance, created, **kwargs):
-            pass
-
-        post_save.connect(post_save_receiver, sender=settings.AUTH_USER_MODEL)
-
-    Generally speaking, you should reference the User model with the
-    :setting:`AUTH_USER_MODEL` setting in code that is executed at import
-    time. ``get_user_model()`` only works once Django has imported all models.
+        It is now possible to call ``get_user_model()`` while Django is
```

Comment:
I was thinking that relations might be cached, but I guess `apps.clear_cache()` in the signal handler takes care of that.

I was really surprised that models.`ForeignKey(get_user_model(), ...` generates a correct migration with `to=settings.AUTH_USER_MODEL`. Not sure how that works offhand!

p.s. There's some suggested wording for this note in my first comment.